_posts/2021-03-10-Bootable-jar-jkube-clustering-dns-ping-openshift.adoc

```diff
@@ -0,0 +1,162 @@
+====
+
+== Introduction
+
+We recently described in this https://www.wildfly.org/news/2021/02/01/Bootable-jar-jkube-clustering-openshift[post] details about how to deploy and configure a WildFly Bootable JAR cluster application on OpenShift. By default, a Bootable JAR cluster application uses the http://www.jgroups.org/manual5/index.html#_kube_ping[KUBE_PING] protocol for clustering discovery mechanism. This protocol is based on the ability to query the Kubernetes API to determine which pods match certain criteria and then create a JGroups cluster based on this information. This protocol could not be a viable or an ideal solution for your scenario, for example, if you cannot grant a specific permission on your Kubernetes cluster to use KUBE_PING, or if your requirement is to make your application agnostic to the Kubernetes API.
+
+In this guide we will explain how you can configure the same Bootable JAR cluster application but this time by using the http://www.jgroups.org/manual5/#_dns_ping[DNS_PING] protocol.
+
+== Getting started
+
+The following are the key points we need to accomplish to configure our Bootable JAR cluster application to use DNS_PING:
+
+* *Define a ping service.* We need to create a https://kubernetes.io/docs/concepts/services-networking/service/#headless-services[headless service] resource, which is used by the protocol to get the IP addresses of the pods that would be part of the cluster. When we are defining this resource, we need to pay attention to the service selector and service name. The selector defines the pods that will be backed by this service. The JGroups subsystem needs to know the service name to configure the DNS_PING protocol.
+
+* *Configure the JGroups subsystem to use DNS_PING.* We need to change the default JGroups subsystem configuration of our Bootable JAR application. We will execute a CLI script at build time to configure the JGroups protocol stack, removing the default KUBE_PING protocol and adding the DNS_PING. One important detail about the DNS_PING configuration is the value we will use for the *dns_query* property. This value specifies the DNS name of the Kubernetes ping service the protocol is going to use to discover other cluster members.
+
+To sum up, we will need a Kubernetes service with a matching label selector to define which pods will be part of our cluster, and we will tell to the JGroups subsystem that we want to use such a service to locate other possible cluster members.
+
+== Define a ping service
+
+The ping service is just an additional Kubernetes service resource. We are using https://www.eclipse.org/jkube/docs/openshift-maven-plugin[JKube openshift-maven-plugin] to deploy our application, this maven plugin allows us to specify any additional resource we want to deploy. What we need to do is to create our additional resources under *src/main/jkube/* and JKube will create the resource for us when deploying the application.
```

Comment:
s/are using/are using the/g

s/application, this/application. This/g

s/is to create/is create/g